openstack/compute/v2/extensions/lockunlock/requests.go

```diff
@@ -0,0 +1,19 @@
+package lockunlock
+
+import "github.com/gophercloud/gophercloud"
+
+func actionURL(client *gophercloud.ServiceClient, id string) string {
+	return client.ServiceURL("servers", id, "action")
+}
+
+// Lock is the operation responsible for locking a Compute server.
+func Lock(client *gophercloud.ServiceClient, id string) (r gophercloud.ErrResult) {
```

Comment:
Instead of `gophercloud.ErrResult`, let's change this to `LockResult`. I believe the reason to do this is to localize the result strictly to the Lock functionality. This is _not_ done in the `pauseunpause` extension that I referenced in #518, which means `pauseunpause` should be updated to reflect this as well. See [here](https://github.com/gophercloud/gophercloud/pull/444#pullrequestreview-57964997) for more discussion. As well, you can use the `migrate` action as a template for how to implement the response.